README.md

```diff
@@ -1,2 +1,228 @@
-# satellite6-upgrade
-Satellite 6 upgrade related tools
+# Satellite6-upgrade
+
+Satellite6 upgrade is a set of tools to be used for following reasons:
+1. Creating and deleting satellite/capsule instances using RHEVM templates/Openstack images
+2. Upgrade Satellite6, Capsule and clients (RHEVM instances / Users own setup)
+3. Post Upgrade Satellite Entities Verification
+
+## Installation
+
+ Satellite6-Upgrade depends on ```pycurl``` being installed, but installing it,
+ specially on a virtual environment is not straight forward. You must run the
+ following script in order to have ```pycurl``` installed properly.
+
+##### Make sure pip is updated
+
+    pip install -U pip
+
+##### Install pycurl using the proper cryptography library
+
+    if [ "$(curl --version | grep NSS 2>/dev/null)" ]; then
+        pip install --compile --install-option="--with-nss" pycurl
+    else
+        pip install --compile --install-option="--with-openssl" pycurl
+    fi
+
+Finally, python packages listed in ```requirements.txt``` must be installed before
+satellite6-upgrade can be used:
+
+    pip install -r requirements.txt
+
+## Usage Examples
+
+### Create Satellite/Capsule Instance from RHEVM template
+To create a live satellite/capsule instance from RHEVM template:
+*Pre-Conditions: The template should have a satellite/capsule populated in it already.*
+
+    RHEV_USER=<user@domain> RHEV_PASSWD=<passwd> RHEV_URL='https://<rhev_url:port>/api'
+    fab create_rhevm_instance:<new_instance_name>,<template_name>,<datacenter>,<quota>,
+    <cluster>,<timeout_in_minutes>
+
+### Delete Satellite/Capule Instance from RHEVM template
+To delete a satellite/capsule instance from RHEVM template:
+*Pre-Condition - The instance should be already created*
+
+    RHEV_USER=<user@domain> RHEV_PASSWD=<passwd> RHEV_URL='https://<rhev_url:port>/api'
+    fab delete_rhevm_instance:<new_instance_name>,<timeout_in_minutes>
+
+### Setup, Ready and Upgrade the Satellite/Capsule/Clients
+As we look in previous examples, the command is consisted of mainly two parts that is
+1. ```Environment variables```
+2. ```fab command```
+
+In this section, we will learn each part separately.
+
+We need different environment variables and fab command depends on following Upgrade Types:
+1. Satellite
+2. Capsule
+3. Clients
+4. Longrun
+
+Again, environment variables required for each upgrade type has been divided in parts and subparts,
```

Comment:
`Again, environment variables required for each upgrade type have been divided into parts and then the subparts,`